### ISSUE TITLE
Graphql submitting each mutation twice in the same request

### ISSUE THREAD

**Issue by tpendragon** [pulibrary/figgy#3233]
## Backtrace

    line 75 of [PROJECT_ROOT]/app/adapters/instrumented_adapter.rb: block in save
    line 48 of [PROJECT_ROOT]/app/adapters/instrumented_adapter.rb: block in trace
    line 44 of [PROJECT_ROOT]/app/adapters/instrumented_adapter.rb: trace

[View full backtrace and more info at honeybadger.io](https://app.honeybadger.io/projects/53391/faults/48989358)

**Comment by hackmastera** [comment]
Here's a gist of a log. It looks like the mutation is in there twice. If our endpoint runs each one, that would explain stale object errors. https://gist.github.com/hackmastera/4781dd5c4f8b79946fdc996e2003e2c2

**Comment by jrgriffiniii** [comment]
I could only reproduce this by submitting changes for FileSets twice in succession, quickly enough to ensure that the following `Promise` within the `async`/`await` does not resolve for the initial request:
```
try {
      const response = await apollo.mutate({
        mutation, variables
      })
      // reset the state to reflect applied changes
      context.commit('SAVED_STATE', 'SAVED')
      setTimeout(function(){ context.commit('APPLY_STATE') }, 1000);
    } catch(err) {
      context.commit('SAVED_STATE', 'ERROR')
      console.error(err)
    }
```
(https://github.com/pulibrary/figgy/blob/master/app/javascript/store/vuex/actions.js#L107)

While this action is being dispatched, the submit button should be disabled.

**Comment by hackmastera** [comment]
@jrgriffiniii When reproduced that way, do you get the log with mutations in a single request?

**Comment by jrgriffiniii** [comment]
I've spent nearly one hour trying to reproduce this on the `master` branch (perhaps I'm simply running too many processes in the background to trigger the race condition?).  I've tried modifying the code in order to ensure that multiple GraphQL requests are sent, simultaneously, without using async/await or explicit promises.

I'll attempt to reproduce this again once I've cleared my environment for better performance.
